// openweathermap-plugin/WeatherFunctions.js
var fs = require("fs");
var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;

function WeatherFunction(){

	this.stateMap={};
	this.airPortMap={};
	this.fileName="./capitals.txt";
	this.APIKeys=["b323832de00f2c2f","2c50b81f5282351b", "828696a5aae9b279","58062bbbe2add910", "62cc417d1996d567", "e157d6553575df8b", "7bdee4aaa7cae4c4" ];
};

WeatherFunction.prototype={

	getTemperatureForAll: function(){
		var i=0;
		var temperatureArray = [];
		for(var state in this.stateMap){
			if(i>=7){
				i=0;
			}
			temperatureArray.push({state:state,temp:this.getTemperatureUsingKey(state, this.APIKeys[i])});
			i=i+1;
		}
		return temperatureArray;

	},

	getTestData : function(){
		var j=0;
		for(var state in this.stateMap){
			
			for(i=1; i<=12; i++){
				var fileName = "./history/city" + j + "_Month" + i + ".txt";
				fs.writeFile(fileName, this.getJSON("http://www.wunderground.com/history/airport/" + this.airPortMap[state] + "/2014/" + i + "/1/MonthlyHistory.html?req_city=" + this.stateMap[state] + "&req_state=" + state + "&reqdb.magic=1&reqdb.wmo=99999&format=1"));
			}
			j = j+1;
			console.log("Finished creating files for state:" + state);
		}
	},

	createStateDictionary: function (){
		var array = fs.readFileSync(this.fileName).toString().split("\n");
		for(i in array) {
		    var splitArray = array[i].split(",");
		    this.stateMap[splitArray[0]] = splitArray[1];
		    this.airPortMap[splitArray[0]] = splitArray[2];
		}
	},

	getJSON: function(data){
		var jsonReq = new XMLHttpRequest();// a new request
		jsonReq.open("GET",data,false);
		jsonReq.send(null);
		return jsonReq.responseText;          
	},

	getTemperatureUsingKey: function(stateName, key){
		//console.log("http://api.wunderground.com/api/" + key + "/conditions/q/" + stateName + "/" + this.stateMap[stateName] + ".json");
		var json_obj = JSON.parse(this.getJSON("http://api.wunderground.com/api/" + key + "/conditions/q/" + stateName + "/" + this.stateMap[stateName] + ".json"));
		return json_obj.current_observation.temp_c;
	},

	//Need to add the API key
	getTemperature: function(stateName){

		var json_obj = JSON.parse(this.getJSON("http://api.wunderground.com/api/b323832de00f2c2f/conditions/q/" + stateName + "/" + this.stateMap[stateName] + ".json"));
		return json_obj.current_observation.temp_c;
	},

	getHumidityUsingKey: function(stateName, key){
		var json_obj = JSON.parse(this.getJSON("http://api.wunderground.com/api/" + key + "/conditions/q/" + stateName + "/" + this.stateMap[stateName] + ".json"));
			return json_obj.current_observation.relative_humidity;
	},

	getHumidity: function (stateName){

		var json_obj = JSON.parse(this.getJSON("http://api.wunderground.com/api/b323832de00f2c2f/conditions/q/" + stateName + "/" + this.stateMap[stateName] + ".json"));
		return json_obj.current_observation.relative_humidity;
	},

	getWindUsingKey: function(stateName, key){
		var json_obj = JSON.parse(this.getJSON("http://api.wunderground.com/api/" + key + "/conditions/q/" + stateName + "/" + this.stateMap[stateName] + ".json"));
			return json_obj.current_observation.wind_mph;
	},

	getWind: function (stateName){
		var json_obj = JSON.parse(this.getJSON("http://api.wunderground.com/api/b323832de00f2c2f/conditions/q/" + stateName + "/" + this.stateMap[stateName] + ".json"));
		return json_obj.current_observation.wind_mph;
	},
};

module.exports = WeatherFunction;
