// openweathermap-plugin/WeatherFunctions.js
var fs = require("fs");
var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;

function WeatherFunction(){

	this.stateMap={};
	this.fileName="./capitals.txt";
};

WeatherFunction.prototype={

	createStateDictionary: function (){
		var array = fs.readFileSync(this.fileName).toString().split("\n");
		for(i in array) {
		    var splitArray = array[i].split(",");
		    this.stateMap[splitArray[0]] = splitArray[1];
		}
	},

	getJSON: function(data){
		var jsonReq = new XMLHttpRequest();// a new request
		jsonReq.open("GET",data,false);
		jsonReq.send(null);
		return jsonReq.responseText;          
	},

	//Need to add the API key
	getTemperature: function(stateName){

		var json_obj = JSON.parse(this.getJSON("http://api.openweathermap.org/data/2.5/weather?q=" + this.stateMap[stateName]));
		return json_obj.main.temp;
	},

	getHumidity: function (stateName){

		var json_obj = JSON.parse(this.getJSON("http://api.openweathermap.org/data/2.5/weather?q=" + this.stateMap[stateName]));
		return json_obj.main.humidity;
	},

	getWind: function (stateName){
		var json_obj = JSON.parse(this.getJSON("http://api.openweathermap.org/data/2.5/weather?q=" + this.stateMap[stateName]));
		return json_obj.wind.speed;
	},
};

module.exports = WeatherFunction;
