// openweathermap-plugin/WeatherFunctions.js
var fs = require("fs");
var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;

function WeatherFunction(){

	this.stateMap={};
	this.fileName="./capitals.txt";
};

WeatherFunction.prototype={

	createStateDictionary: function (){
		var array = fs.readFileSync(this.fileName).toString().split("\n");
		for(i in array) {
		    var splitArray = array[i].split(",");
		    this.stateMap[splitArray[0]] = splitArray[1];
		}
	},

	getJSON: function(data){
		var jsonReq = new XMLHttpRequest();// a new request
		jsonReq.open("GET",data,false);
		jsonReq.send(null);
		return jsonReq.responseText;          
	},

	//Need to add the API key
	getTemperature: function(stateName){

		var json_obj = JSON.parse(this.getJSON("http://api.wunderground.com/api/b323832de00f2c2f/conditions/q/" + stateName + "/" + this.stateMap[stateName] + ".json"));
		return json_obj.current_observation.temp_c;
	},

	getHumidity: function (stateName){

		var json_obj = JSON.parse(this.getJSON("http://api.wunderground.com/api/b323832de00f2c2f/conditions/q/" + stateName + "/" + this.stateMap[stateName] + ".json"));
		return json_obj.current_observation.relative_humidity;
	},

	getWind: function (stateName){
		var json_obj = JSON.parse(this.getJSON("http://api.wunderground.com/api/b323832de00f2c2f/conditions/q/" + stateName + "/" + this.stateMap[stateName] + ".json"));
		return json_obj.current_observation.wind_mph;
	},
};

module.exports = WeatherFunction;
